test/libsolidity/smtCheckerTests/external_calls/external_hash_known_code_state_reentrancy.sol

```diff
@@ -0,0 +1,32 @@
+pragma experimental SMTChecker;
+
+contract State {
+	uint x;
+	C c;
+	function f() public view returns (uint) {
+		return c.g();
+	}
+}
+
+contract C {
+	address owner;
+	uint y;
+	State s;
+
+	constructor() public {
+		owner = msg.sender;
+	}
+
+	function f() public view {
+		address prevOwner = owner;
+		uint z = s.f();
+		assert(z == y);
+		assert(prevOwner == owner);
+	}
+
+	function g() public view returns (uint) {
+		return y;
+	}
+}
+// ----
+// Warning: (306-320): Assertion violation happens here
```

Comment:
Which one is this?

Reply:
z == y

Reply:
it fails because `State.c` is not necessarily the caller

Reply:
and since the state is not carried over to the called contract, it doesn't know anything about `c`'s state